sources/assets/Stride.Core.Assets.Quantum/AssetToBaseNodeLinker.cs

```diff
@@ -48,7 +48,13 @@ public override ObjectReference FindTargetReference(IGraphNode sourceNode, IGrap
             var targetIds = CollectionItemIdHelper.GetCollectionItemIds(targetNode.Retrieve());
             var itemId = sourceIds[sourceReference.Index.Value];
             var targetKey = targetIds.GetKey(itemId);
-            return targetReference.FirstOrDefault(x => Equals(x.Index.Value, targetKey));
+            foreach (var targetRef in targetReference)
+            {
+                if (Equals(targetRef.Index.Value, targetKey))
+                    return targetRef;
+            }
+
+            return null;
```

Comment:
That's exactly what `FirstOrDefault` does and I find it easier to read.

Reply:
This is the hot path, those methods are called millions of times a second when loading assets. All of those FirstOrDefault allocation adds up and trashes the GC causing significant slowdown for any new allocation. All of those changes are not baseless, I profiled them thoroughly.